### ISSUE TITLE
Sync Fails Due to Signing Root Verification

### ISSUE THREAD

**Issue by nisdas** [prysmaticlabs/prysm#3733]
```
[2019-10-07 06:20:43] DEBUG forkchoice: Executing state transition on block slot=101313
panic: could not process block from fork choice service: could not execute state transition: could not process block: could not process block header: parent root 0xf1d55f237029608bbf68f2bc735d4f65f6904a48e1ca5c6fc896d0ee1e540ad5 does not match the latest block header signing root in state 0x8b3957fcb00be9fc686156618f7e2ca37f1adec8e83926d3c968ef488450920d
```

The same bug pops up even after multiple restarts and occurs at the same block slot of `101313`

**Comment by stefa2k** [comment]
Got the same issue

```
[2019-10-07 13:02:54]  INFO initial-sync: Processing block 101311/118747 - estimated time remaining 45m3s blocksPerSecond=6.5 peers=5/7
[2019-10-07 13:02:54]  INFO initial-sync: Processing block 101312/118747 - estimated time remaining 46m52s blocksPerSecond=6.2 peers=5/7
[2019-10-07 13:02:54]  INFO initial-sync: Processing block 101313/118747 - estimated time remaining 46m29s blocksPerSecond=6.2 peers=5/7
panic: could not process block from fork choice service: could not execute state transition: could not process block: could not process block header: parent root 0xf1d55f237029608bbf68f2bc735d4f65f6904a48e1ca5c6fc896d0ee1e540ad5 does not match the latest block header signing root in state 0x8b3957fcb00be9fc686156618f7e2ca37f1adec8e83926d3c968ef488450920d

goroutine 177 [running]:
github.com/prysmaticlabs/prysm/beacon-chain/sync/initial-sync.(*InitialSync).Start(0xc0009122d0)
        beacon-chain/sync/initial-sync/service.go:107 +0x98d
created by github.com/prysmaticlabs/prysm/shared.(*ServiceRegistry).StartAll
        shared/service_registry.go:44 +0x23e
```

**Comment by Skretch** [comment]
I have the same issue at block 26073

```
[2019-10-10 10:22:34]  INFO initial-sync: Processing block 26071/32422 - estimated time remaining 4m36s blocksPerSecond=23.0 peers=4/5
[2019-10-10 10:22:34]  INFO initial-sync: Processing block 26072/32422 - estimated time remaining 4m35s blocksPerSecond=23.1 peers=4/5
[2019-10-10 10:22:35]  INFO initial-sync: Processing block 26073/32422 - estimated time remaining 4m34s blocksPerSecond=23.1 peers=4/5
panic: could not process block from fork choice service: could not execute state transition: could not process block: could not process block header: parent root 0x77300337775820962d13304c27c9274e5926cbd39b245661e2f21906e177378a does not match the latest block header signing root in state 0xf261c8e363f35d95b1d071cfc9d35539a0dc9da957a7a567fe2c9e2f5081aa8b

goroutine 177 [running]:
github.com/prysmaticlabs/prysm/beacon-chain/sync/initial-sync.(*InitialSync).Start(0xc0006fbb60)
        beacon-chain/sync/initial-sync/service.go:107 +0x98d
created by github.com/prysmaticlabs/prysm/shared.(*ServiceRegistry).StartAll
        shared/service_registry.go:44 +0x23e
```

**Comment by stefa2k** [comment]
@Skretch yes I can confirm I got the same error now at the same block.

**Comment by Doppelbock42** [comment]
I'm getting this also.
`[2019-10-10 15:42:34]  INFO initial-sync: Processing block 26072/35622 - estimated time remaining 2h7m20s blocksPerSecond=1.2 peers=4/6
[2019-10-10 15:42:34]  INFO initial-sync: Processing block 26073/35622 - estimated time remaining 2h2m25s blocksPerSecond=1.3 peers=4/6
panic: could not process block from fork choice service: could not execute state transition: could not process block: could not process block header: parent root 0x77300337775820962d13304c27c9274e5926cbd39b245661e2f21906e177378a does not match the latest block header signing root in state 0xf261c8e363f35d95b1d071cfc9d35539a0dc9da957a7a567fe2c9e2f5081aa8b

goroutine 169 [running]:
github.com/prysmaticlabs/prysm/beacon-chain/sync/initial-sync.(*InitialSync).Start(0xc00096d320)
        beacon-chain/sync/initial-sync/service.go:107 +0x98d
created by github.com/prysmaticlabs/prysm/shared.(*ServiceRegistry).StartAll
        shared/service_registry.go:44 +0x23e`

**Comment by MartinNBG** [comment]
I have the same issue at the same block... ._.

**Comment by rauljordan** [comment]
This is resolved as of the #3865 and latest release
